Source/Plugin.BLE.Android/Adapter.cs

```diff
@@ -184,6 +185,34 @@ public override List<IDevice> GetSystemConnectedOrPairedDevices(Guid[] services
             return connectedDevices.Union(bondedDevices, new DeviceComparer()).Select(d => new Device(this, d, null, 0)).Cast<IDevice>().ToList();
         }
 
+        protected override async Task<IDevice> ConnectNativeAsync(string friendlyName, Guid uuid, Func<IDevice, bool> deviceFilter, CancellationToken cancellationToken = default)
+        {
+            if (uuid == Guid.Empty)
+            {
+                var taskCompletionSource = new TaskCompletionSource<IDevice>();
+                EventHandler<DeviceEventArgs> handler = (sender, args) =>
+                {
+                    taskCompletionSource.TrySetResult(args.Device);
+                };
+
+                DeviceDiscovered += handler;
```

Comment:
Is this ever unhooked?

Reply:
I guess this could be unhooked right after `taskCompletionSource.TrySetResult(args.Device);` is called, if the handler is declared before the lambda expression.

Reply:
Nope it is not 😇. It is fixed